felix/cmd/calico-bpf/commands/policy_debug.go

```diff
@@ -87,6 +89,32 @@ func printInsn(cmd *cobra.Command, insn asm.Insn) {
 	cmd.Println()
 }
 
+func getRuleMatchID(comment string) (uint64, error) {
+	matchID := strings.Split(comment, "Rule MatchID:")[1]
+	matchID = strings.Trim(matchID, " ")
+	id, err := strconv.ParseUint(matchID, 0, 64)
+	if err != nil {
+		return 0, err
+	}
+	return id, nil
+}
+
+func getRuleCounter(comment string) (uint64, error) {
+	matchId, err := getRuleMatchID(comment)
+	if err != nil {
+		return 0, err
+	}
+	rmap := counters.PolicyMap(&bpf.MapContext{})
+	m, err := counters.LoadPolicyMap(rmap)
```

Comment:
Do we need to load the policy map every time we query a single ID? I wonder if it would be just good to read the map once and then just query it. I do not think we are really aiming at the most recent accuracy, we want to get a rough approximation. Otherwise we would need to stop the dp from processing packets anyway.

Reply:
Or we can just always querry the map directly without loading it into memory, but that may be trickier as it is per-cpu map :thinking:

Reply:
Querying the map will be tricky.